### ISSUE TITLE
Not a issue but a request

### ISSUE THREAD

**Issue by Cads182** [Paranoid-AF/SvenCoop_Plugins#3]
Hi Paranoid, id like to start using this plugin on my servers in order to replace **BuyMenu**, to do so can you add the following features please:

1. After buy something, automatically close the menu, just to prevent unintentional buys.
2. Somehow find a way to prevent players to buy an weapon that already had, if so, buy ammo instead for a lower price of course.
3. Despawn time on weapons and items, just to prevent overflow of models.
4. Try to instead of creating the entity in the player location, give it, I mean if the player is in a movable object like (func_train) and jump they could lost the weapon.
5. Revive item, when player die he could buy his respawn, just to prevent weapon lost.
6. Allow buying by names, for example: **!buy crowbar**
7. Have tracking of all players buys, so you can have statistics about most buyed item etc.
8. This is a crazy idea that I had idk if its possible, set a category that only specific STEAMID's could use and buy different stuff for a lower price or special weapons.
9. Add sounds when purchase failed/succeed.

Bugs that I found so far:

1. Sometimes the score on scoreboard doesn't update after a buy, but in the HUD does, looks like the plugin has his own score store system (?) I think that the best thing is directly interact with player frags, just to prevent confusion.

I know that I'm requesting a lot of features, but this could be really usefull for everyone ^^, anyways thank you so much for this very cool plugin!

**Comment by Paranoid-AF** [comment]
Hi there!

First things first, thank you for your attention, it's always a pleasure to hear someone who cares about my work!

Alright, let's get straight to the point. I assume that you're talking about **BuyStuff**, right? I've already made an alternative to it, which is called [**Ecco**](https://github.com/Paranoid-AF/Ecco), and there are already some server running it.

However, **it still does not have those functions that you just asked**. And sadly, it's somehow abandoned because some designs are considered to be outdated and unbearable to myself (which is part of the reason that it still doesn't have an English document). Not to mention that I don't have enough time to work on it, especially when it's something that I don't intend to do. Another reason is that I don't find Sven Co-op fun anymore.

In short: the upgrade sucks, Sven Co-op is gradually becoming less fun to me, and I am always running out of time.

But I'm not here trying to disappoint anyone. As a hobby it's still very fun to me, and recently I just tried out GitHub Actions, and got some inspiration from it. Hopefully I will work on another alternative when I find myself having tons of time and energy. Sure thing that it will include those features that you asked.

Sorry that these are not going to happen very soon, as things are not going very well to me recently.

Stay safe out there, watch out for the virus.
Paranoid_AF

**Comment by Cads182** [comment]
Hey!, to be honest I did not expect such a quick response, I thought you had completely abandoned everything related to Sven Co-op, I'm glad you answered me, in fact I tried Ecco but I could not fully understand it's installation due to the language, regarding your opinion about the game, I do not agree that much, I started playing Sven since 2016 and I was never get bored, in fact I have almost 5.5k hours in the game and I never get bored, I believe that with your knowledge you can achieve many interesting and entertaining things, the thing is to have people to play with, it is understandable, I hope you going well, I appreciate that you have responded, that is enough for me, thanks for your plugins, I hope you take care of yourself, luck!
(I hope you don't mind that I'm constantly editing the post to report some bugs/features 😅)

**Comment by Tucumanos** [comment]
Please don't abandon this plugin
